annotated_test.go

```diff
@@ -1474,6 +1474,159 @@ func TestAnnotate(t *testing.T) {
 	})
 }
 
+func TestAnnotateApply(t *testing.T) {
+	t.Parallel()
+
+	type a struct{}
+	type b struct{ a *a }
+	newA := func() *a { return &a{} }
+	newB := func(a *a) *b {
+		return &b{a}
+	}
+
+	var (
+		errTagSyntaxSpace            = ` multiple tags are not separated by space`
+		errTagKeySyntax              = " tag key is invalid, Use group, name or optional as tag keys"
+		errTagValueSyntaxQuote       = `tag value should start with double quote. Did you mean `
+		errTagValueSyntaxEndingQuote = `tag value should end in double quote. Did you mean `
+		errTagSyntaxColon            = `tag key and value not separated by colon. Did you mean `
+		errTagValueSpace             = `tag key value separated by extra space. Use format` + " `tag_key:\"value\"`"
+	)
+	tests := []struct {
+		give           string
+		wantErr        string
+		giveAnnotation fx.Annotation
+	}{
+		{
+			give:           "ParamTags wrong ending quote",
+			wantErr:        errTagValueSyntaxEndingQuote + "`name:\"something\"` ?",
+			giveAnnotation: fx.ParamTags(`name:"something'`),
+		},
+		{
+			give:           "ParamTags wrong starting quote",
+			wantErr:        errTagValueSyntaxQuote + "`optional:\"true\"` ?",
+			giveAnnotation: fx.ParamTags(`name:"something" optional:'true"`),
+		},
+		{
+			give:           "ParamTags key not made of alphabets",
+			wantErr:        errTagKeySyntax,
+			giveAnnotation: fx.ParamTags(`a"me1:"something"`),
+		},
+		{
+			give:           "ParamTags multiple tags not separated by space",
+			wantErr:        errTagSyntaxSpace,
+			giveAnnotation: fx.ParamTags(`name:"something"group:"something"`),
+		},
+		{
+			give:           "ParamTags key value not separated by colon",
+			wantErr:        errTagSyntaxColon + "`group:\"something\"` ?",
+			giveAnnotation: fx.ParamTags(`group"something"`),
+		},
+		{
+			give:           "ParamTags key not equal to group, name or optional",
+			wantErr:        errTagKeySyntax,
+			giveAnnotation: fx.ParamTags(`name1:"something"`),
+		},
+		{
+			give:           `ParamTags invalid space after key colon`,
+			wantErr:        errTagValueSpace,
+			giveAnnotation: fx.ParamTags(`name: "something"`),
+		},
+		{
```

Comment:
The set of tests for ResultTags looks similar to for ParamTags, can we de-duplicate these tests?